File: benchmarks/cpp/linked_list.cpp
#include <cstdio>
struct Node { long long val; Node* nxt; };
int main() {
    const int n = 50000;
    Node* head = new Node{0, nullptr};
    Node* cur = head;
    for (int i = 1; i < n; i++) { cur->nxt = new Node{i, nullptr}; cur = cur->nxt; }
    long long total = 0;
    for (cur = head; cur; cur = cur->nxt) total += cur->val;
    printf("%lld\n", total);
}

File: benchmarks/linked_list.py
n = 50000
head = {"val": 0, "nxt": None}
cur = head
for i in range(1, n):
    node = {"val": i, "nxt": None}
    cur["nxt"] = node
    cur = node
total = 0
cur = head
while cur is not None:
    total += cur["val"]
    cur = cur["nxt"]
print(total)
